### DISCUSSION TITLE
Godot-bevy extension

### DISCUSSION THREAD

**Discussion by darthLeviN** [bevyengine/bevy#13992] [Q&A]
I'm a godot developer, and bevy's esc is great. So i got this idea that it's so absurd that makes me tear up.
What if we put bevy inside godot?
Is this feasible? It could totally change what godot and bevy are. Godot would turn into a parallelization powerhouse, and bevy would turn into a tool extremely easy to use.

**Comment by alice-i-cecile** [comment]
Yeah, I've seen teams doing this! You'll want to start with [godot-rust](https://godot-rust.github.io/), and focus on `bevy_ecs` + `bevy_app`.

**Reply by jonathandw743** [reply]
does this lose the more fine-grain control that bevy gives?
ie can you do all the same things in godot-rust and bevy or is it rust functions for all the stuff available in godot?

**Reply by alice-i-cecile** [reply]
Yeah, you have the full set of things available in godot-rust + bevy, minus some niceties around things like rendering or input mapping that overlap with what Godot itself provides :)

**Reply by darthLeviN** [reply]
> does this lose the more fine-grain control that bevy gives? ie can you do all the same things in godot-rust and bevy or is it rust functions for all the stuff available in godot?

In godot you can replace your main loop with a custom one without recompiling the engine.
It should be possible to let bevy control the main loop of godot itself.
It's also possible to replace the physics engine without recompilation of godot.
Rendering engine will probably be godot's. 
It's debatable on how to use godot's asset loader instead of bevy's.
You can track "changed" transforms at the end of each iteration and synchronize bevy's world with godot's. this synchronization would be the biggest downside of combining bevy with godot.
